src/gui/properties/pieceavailabilitybar.cpp

```diff
@@ -67,23 +67,17 @@ QVector<float> PieceAvailabilityBar::intToFloatVector(const QVector<int> &vecin,
     // image.x(1) = pieces.x(1.7 >= x < 3.4)
 
     for (int x = 0; x < reqSize; ++x) {
-        // don't use previously calculated value "ratio" here!!!
-        // float cannot save irrational number like 7/9, if this number will be rounded up by std::ceil
-        // give you x2 == pieces.size(), and index out of range: pieces[x2]
-        // this code is safe, so keep that in mind when you try optimize more.
-        // tested with size = 3000000ul
-
         // R - real
-        const float fromR = (x * vecin.size()) / (float)reqSize;
-        const float toR = ((x + 1) * vecin.size()) / (float)reqSize;
+        const float fromR = x * ratio;
```

Comment:
Just to confirm you have read the comment "don't use previously calculated value "ratio" here", right? And I suppose you deemed it ok to use ratio, right?
I am asking just to confirm before possible mergin this.